packages/devextreme/testing/tests/DevExpress.ui.widgets/listParts/commonTests.js

```diff
@@ -1711,6 +1719,42 @@ QUnit.module('options changed', moduleSetup, () => {
     });
 });
 
+QUnit.module('focus on deleting', () => {
+    const items = ['item 1', 'item 2', 'item 3'];
+    QUnit.test('first item should not be focused after deleting when focusStateEnabled = false (T1226030)', function(assert) {
+        const $element = $('#list').dxList({
+            items,
+            focusStateEnabled: false,
+            allowItemDeleting: true,
+            itemDeleteMode: 'static',
+        });
+        let $listItems = $element.find(toSelector(LIST_ITEM_CLASS));
+        const firstItemIsFocused = $listItems.eq(0).hasClass(`${LIST_FOCUS_STATE_CLASS}`);
+
+        $listItems.eq(0).find(`.${STATIC_DELETE_BUTTON_CLASS}`).trigger('dxclick');
+        $listItems = $element.find(toSelector(LIST_ITEM_CLASS));
+
+        assert.notOk(firstItemIsFocused, 'Next Item is not focused');
+    });
+
+    QUnit.test('first item should be focus when deleting when focusStateEnabled = true (T1226030)', function(assert) {
+        const $element = $('#list').dxList({
+            items,
+            focusStateEnabled: true,
+            allowItemDeleting: true,
+            itemDeleteMode: 'static',
+        });
+        let $listItems = $element.find(toSelector(LIST_ITEM_CLASS));
+
+        $listItems.eq(0).find(`.${STATIC_DELETE_BUTTON_CLASS}`).trigger('dxclick');
+        $listItems = $element.find(toSelector(LIST_ITEM_CLASS));
```

Comment:
why do u do it again?

Reply:
The answer is here: https://github.com/DevExpress/DevExtreme/pull/27230#discussion_r1590903102